pkg/gadgets/profile/tcprtt/tracer/tracer_test.go

```diff
@@ -0,0 +1,722 @@
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamByRemoteAddress).Set("true")
+				params.Get(ParamFilterRemoteAddress).Set(serverIP.String())
+				return params
+			},
+			expectedResult: expectedResult{
+				exactHistograms: 1,
+				byRemoteAddr:    serverIP.String(),
+			},
+		},
+		// TODO: Test mix cases with clients and servers with different IPs
+	} {
+		test := test
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			// Create context
+			gadgetCtx := newGadgetCtx(test.getGadgetParams(), test.timeout)
+			defer gadgetCtx.Cancel()
+
+			// Run the tracer in a goroutine
+			tracer := createTracer()
+			out := make(chan []byte)
+			go func() {
+				r, err := tracer.RunWithResult(gadgetCtx)
+
+				// Validate error. Use assert instead of require to avoid
+				// blocking the main goroutine.
+				if test.expectedResult.err {
+					assert.NotNil(t, err, "expected error running tracer")
+				} else {
+					assert.Nil(t, err, "not expected error running tracer")
+				}
+
+				out <- r
+			}()
+
+			// Wait for the tracer to be ready
+			time.Sleep(2 * time.Second)
+
+			// Generate the events
+			runner := utilstest.NewRunnerWithTest(t, runnerConfig)
+			utilstest.RunWithRunner(t, runner, func() error {
+				connectTCPClient(t, clientIP, serverIP, serverPort)
+				return nil
+			})
+
+			// If needed, stop the tracer. Otherwise, simply wait for the result
+			if test.timeout == 0 {
+				// Wait for the tracer to capture the events
+				time.Sleep(2 * time.Second)
+
+				// Stop the tracer
+				gadgetCtx.Cancel()
+			}
+
+			// Get the result. If we expect an error, we don't need to validate
+			data := <-out
+			if test.expectedResult.err {
+				require.Nil(t, data, "available data when error occurred")
+				return
+			}
+
+			// Unmarshal the result
+			var result types.Report
+			err := json.Unmarshal(data, &result)
+			require.Nil(t, err, "unmarshalling report")
+			lHis := len(result.Histograms)
+
+			// Dump the result
+			for i, l := range result.Histograms {
+				t.Logf("Result [%d/%d]:", i+1, lHis)
+				t.Logf("AddrType %s - Addr %s - Avg %f", l.AddressType, l.Address, l.Average)
+				t.Logf("Histogram: %s", l.Histogram)
+			}
+
+			// Validate the number of histograms
+			if test.expectedResult.exactHistograms != 0 {
+				require.Equal(t, test.expectedResult.exactHistograms, lHis,
+					"invalid number of histograms")
+			} else {
+				// When we don't filter by address, we can't know how many
+				// histograms will be generated. This is because the number of
+				// histograms depends on the number of connections that are
+				// established in the host during the test. Therefore, we can
+				// only check that the number of histograms is greater or equal
+				// than the minimum number of histograms we expect.
+				require.GreaterOrEqual(t, lHis, test.expectedResult.minHistograms,
+					"expecting at least %d histograms, got %d", test.expectedResult.minHistograms, lHis)
```

Comment:
nit: the testing library already prints a `"X" is not greater than or equal to "Y"` message when the test fails. I think just printing a "wrong number of histograms" or similar message is enough.

Reply:
Yep, I updated all the message after I discovered that and maybe I missed some.